javaslang/src/test/java/javaslang/MatchTest.java

```diff
@@ -219,6 +219,26 @@ public void shouldDecomposeSomeSomeTuple() {
 
     // -- List
 
+    @Test
+    public void shouldDecomposeEmptyList() {
```

Comment:
> An empty PriorityQueue has no head

An empty `List` doesn't have a `head` either. If we can decompose a `List`, we should be able to decompose any `Traversable` - it should be possible to base it on `isEmpty` or to provide separate `empty` classes for (FYI: https://github.com/javaslang/javaslang/pull/1300#discussion_r61717723).